File: docs-site/android/close.kt
client.on("chat.room1") { payload -> 
  Log.d("Message", payload)
}

client.on("CONNECTED") {
  // ... Code from previous step

  val unsubscribed = client.off("chat.room1")
  Log.d("SDKEvent", "Unsubscribed from chat.room1 => $unsubscribed")

  client.close()
}
                      
client.connect()

File: docs-site/ios/close.swift
private class ChatMessageListener: MessageListener {
        
    func onMessage(_ data: Any) {
      print("We got a message!")
      print(data)
    }
}

private class SDKEventListener: MessageListener {
        
    func onMessage(_ data: Any) {
      // ... Code from previous step

      let unsubscribe = try await realtime.off(topic: "chat.room1")
      print("Unsubscribed from chat.room1 => \\(unsubscribe)")

      try await realtime.close()
    }
}

var messageListener = ChatMessageListener()
try await realtime?.on(topic: "chat.room1", listener: messageListener!)

var sdkEventListenener = SDKEventListener()
try await realtime?.on(topic: "CONNECTED", listener: sdkEventListenener!)

client.connect()